src/lib/users.ts

```diff
@@ -239,3 +240,46 @@ export async function loadUserCompletedCourses(token?: string): Promise<any> {
     }
   }
 }
+
+export async function loadUserCompletedLessons(token?: string): Promise<any> {
+  const query = gql`
+    query completedCourses {
+      user_progress {
+        collection {
+          ... on Playlist {
+            lessons {
+              slug
+              completed
+            }
+          }
+          ... on Course {
+            lessons {
+              slug
+              completed
+            }
+          }
+        }
+      }
+    }
+  `
+  token = token || getAccessTokenFromCookie()
+  const graphQLClient = getGraphQLClient(token)
+
+  const {user_progress} = await graphQLClient.request(query)
+
+  const completeLessons = flatten(
```

Comment:
here I create the array of slugs of completed lessons from collections that are in progress